### PR TITLE
test(bindparam): e2e coverage for cross-temporal binds

### PR DESCRIPTION
**Pull request by sfc-gh-makowalski**
## Summary

Stacked on top of #978 (`cross-temporal-binds`). Adds e2e coverage in `odbc_tests/tests/e2e/types/` for the four cross-temporal `SQLBindParameter` conversions introduced in #978, exercising the full pipeline against a real Snowflake account: `SQLPrepare → SQLBindParameter → SQLExecute → SELECT → SQLGetData`.

This complements the unit tests in #978 (which exercise the C-struct → JSON parameter binding boundary inside the Rust crate) by verifying actual Snowflake round-trip behavior.

## What's covered

The four new files mirror the existing same-direction test files (`c_date_conversion_to_sql_date.cpp` / `c_time_conversion_to_sql_time.cpp` / `c_timestamp_conversion_to_sql_timestamp.cpp`):

| File | Tests | Direction | Edge cases covered |
|------|-------|-----------|---------------------|
| `c_date_conversion_to_sql_timestamp.cpp` | 6 | `SQL_C_TYPE_DATE → TIMESTAMP_NTZ / _LTZ / _TZ` | leap day (2024-02-29), epoch (1970-01-01), `NULL` indicator |
| `c_timestamp_conversion_to_sql_date.cpp` | 6 | `SQL_C_TYPE_TIMESTAMP → DATE` | nanos discarded, midnight no-info-loss, end-of-day **not** rolled forward, leap day, `NULL` indicator |
| `c_timestamp_conversion_to_sql_time.cpp` | 6 | `SQL_C_TYPE_TIMESTAMP → TIME` | whole-second `h/m/s` preserved, **non-zero `fraction` rejected with SQLSTATE 22008** (Appendix D forbids silent truncation), midnight, end-of-day, epoch-date irrelevance, `NULL` indicator |
| `c_time_conversion_to_sql_timestamp.cpp` | 4 | `SQL_C_TYPE_TIME → TIMESTAMP_NTZ / _LTZ / _TZ` | "current local date" with `[Local::now() before, Local::now() after]` window assertion to handle midnight rollover, `fraction = 0` |

22 tests total. All passed locally against my Snowflake account.

## Spec alignment

These tests verify the behavior prescribed by ODBC Appendix D ("C to SQL: Date / Time / Timestamp") that #978 implements:

- `SQL_C_TYPE_TIMESTAMP → SQL_DATE`: extract date portion (truncate, no rollover).
- `SQL_C_TYPE_TIMESTAMP → SQL_TIME`: extract time portion (`h/m/s` preserved); `SQL_TIME_STRUCT` has no fractional field, so the driver must return `SQL_ERROR` / `SQLSTATE 22008` when the source `fraction` is non-zero rather than silently dropping it.
- `SQL_C_TYPE_DATE → SQL_TIMESTAMP*`: combine date with `00:00:00.000000000`.
- `SQL_C_TYPE_TIME → SQL_TIMESTAMP*`: pair time with current local date and zero fractional seconds.

## Test plan

- [x] Local build of all 4 test targets via `cmake --build odbc_tests/cmake-build --target …` succeeds (Google C++ style, `clang-format`'d, 120-col limit).
- [x] All 22 new tests pass against a Snowflake account.
- [ ] CI runs e2e against the shared test account.

### CODE REVIEW

**Comment by sfc-gh-makowalski** [comment]
* **#1067** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/1067?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1058** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/1058?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1038** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/1038?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#980** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/980?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#992** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/992?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/992?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#978** <a href="https://app.graphite.com/github/pr/snowflakedb/universal-driver/978?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by sfc-gh-makowalski** [comment]
@Copilot review

**Comment by sfc-gh-makowalski** [comment]
@Copilot review

Addressed your 4 comments:

- **DATE → TIMESTAMP_LTZ / _TZ / leap-day _NTZ**: added `CHECK(result.fraction == 0)` in [953cfae12](https://github.com/snowflakedb/universal-driver/pull/992/commits/953cfae12).
- **TIMESTAMP → TIME description mismatch**: updated PR description — tests are correct (Appendix D mandates SQL_ERROR / SQLSTATE 22008 when source `fraction` is non-zero, since `SQL_TIME_STRUCT` has no fractional field), the previous "nanos preserved" wording was misleading and is now "`h/m/s` preserved; non-zero `fraction` rejected with 22008".

File: odbc_tests/tests/e2e/types/c_time_conversion_to_sql_timestamp.cpp [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+// Per ODBC Appendix D ("C to SQL: Time"): "The date fields of the
+// timestamp structure are set to the current date, and the fractional
+// seconds portion of the timestamp is set to zero." The "current date"
+// here is the driver host's local date at bind time.
+//
+// To stay deterministic across midnight rollover, the date assertion
+// captures the local date both before and after the bind and accepts
+// any value within that window.
+
+#include <chrono>
```
[the remark this answers is not in the document]
Comment (sfc-gh-makowalski) [reply odbc_tests/tests/e2e/types/c_time_conversion_to_sql_timestamp.cpp]: Done in 77b8c4b6a — dropped the unused `<chrono>` include (only `<ctime>` is needed for `localtime_r`).

**Comment by sfc-gh-pbulawa** [comment]
Missing gherkins for all of the tests.

**Comment by sfc-gh-pbulawa** [comment]
Let's test the invalid struct fields (e.g. month = 13).

**Review by sfc-gh-pbulawa — COMMENTED** [review]
PTAL at the comments

**Review by sfc-gh-pbulawa — APPROVED** [review]
